### PR TITLE
[WIP] heptio-authenticator-aws side car with apiserver

### PR DESCRIPTION
**Pull request by abhinavdahiya**
This adds https://github.com/heptio/authenticator as a side car with the kube-apiserver. This will allow users to use their aws credentials.

- `aws-authenticator-config.yaml` file configures the authenticator with required AWS resource names to kubernetes group,username

/cc: @ericchiang

### CODE REVIEW

**Comment by abhinavdahiya** [comment]
@ericchiang does this look like a workable path towards integrating this?

File: hack/terraform-quickstart/main.tf [hunk 1]
```diff
@@ -16,17 +16,18 @@ resource "aws_instance" "bootstrap_node" {
...
+  )}"
 
   root_block_device {
     volume_type = "gp2"
     volume_size = "30"
   }
 
   provisioner "file" {
-    source = "environment_${var.environment}.txt"
+    source      = "environment_${var.environment}.txt"
```
Comment (ericchiang) [inline hack/terraform-quickstart/main.tf]: Ah, we should make sure we `terraform fmt` all of these files as part of CI.

cc @rphillips

File: pkg/asset/images.go [hunk 2]
```diff
@@ -2,14 +2,15 @@ package asset
...
+	Flannel:          "quay.io/coreos/flannel:v0.10.0-amd64",
+	FlannelCNI:       "quay.io/coreos/flannel-cni:v0.3.0",
+	Calico:           "quay.io/calico/node:v2.6.6",
+	CalicoCNI:        "quay.io/calico/cni:v1.11.2",
+	Hyperkube:        "gcr.io/google_containers/hyperkube:v1.9.2",
+	KubeDNS:          "gcr.io/google_containers/k8s-dns-kube-dns-amd64:1.14.8",
+	KubeDNSMasq:      "gcr.io/google_containers/k8s-dns-dnsmasq-nanny-amd64:1.14.8",
+	KubeDNSSidecar:   "gcr.io/google_containers/k8s-dns-sidecar-amd64:1.14.8",
+	PodCheckpointer:  "quay.io/coreos/pod-checkpointer:3cd08279c564e95c8b42a0b97c073522d4a6b965",
+	AWSAuthenticator: "quay.io/abhinavdahiya/aws-authenticator:dev",
```
Comment (ericchiang) [inline pkg/asset/images.go]: Why `quay.io/abhinavdahiya`? We shouldn't pull images from an individual developer.

File: pkg/asset/internal/templates.go [hunk 3]
```diff
@@ -184,6 +187,25 @@ spec:
...
+        - mountPath: /var/aws-authenticator
+          name: var-aws-authenticator
+          readonly: true
+      - args:
+        - server
+        - --config=/etc/kubernetes/aws-authenticator/config.yaml
+        image: {{ .Images.AWSAuthenticator }}
+        name: aws-authenticator
+        volumeMounts:
+        - mountPath: /etc/ssl/certs
```
Comment (ericchiang) [inline pkg/asset/internal/templates.go]: Why do we need this?

**Comment by ericchiang** [comment]
Awesome.

We need to think about how we're actually going to merge this, it plays pretty heavily into "what's the role of render?". We probably don't want to add arbitrary flags to render, it needs to remain a reference architecture. Do we want to switch to something like https://github.com/poseidon/terraform-render-bootkube/ ?

I'm going to open an issue describing what we want to do with this TLS bootstrapping attestation work, and how it fits into bootkube. Hopefully we can decide if this a set of experimental flags on bootkube, if it's going to be an internal only thing, if we want to switch to more sophisticated rendering, etc.

File: hack/quickstart/aws-authenticator-config.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# a unique-per-cluster identifier to prevent replay attacks (see above)
```
Comment (ericchiang) [inline hack/quickstart/aws-authenticator-config.yaml]: In the current setup we probably still want bootkube rendering this. Let me open an issue before we address that.

**Pull request closed without merging**